### PR TITLE
[d16-5] null-protect stepTimer.Stop()/Dispose() in DebuggerSession.OnTargetEv…

### PR DESCRIPTION
**Pull request by jstedfast**
…ent()

Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1060768/

Backport of https://github.com/mono/debugger-libs/pull/289

### CODE REVIEW

File: Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs [hunk 1]
```diff
@@ -1196,8 +1196,8 @@ internal protected void OnTargetEvent (TargetEventArgs args)
 				activeThread = args.Thread;
 
 			if (stepTimer != null && (HasExited || args.IsStopEvent)) {
-				stepTimer.Stop (true);
-				stepTimer.Dispose ();
+				stepTimer?.Stop (true);
```
Comment (Therzok) [inline Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs]: But there's a null check right above?

Comment (Therzok) [reply Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs]: Capture stepTimer in a local varaible before it's used, and then you can guarantee it's the same one being used.

Comment (Therzok) [reply Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs]: If this code is multi-threaded, atomic swaps should be used:

```csharp
var timer = Interlocked.Exchange (stepTimer, null);
if (timer != null && ..) {
    timer.Stop (true);
}

```

Comment (jstedfast) [reply Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs]: thanks for the tip!

**Pull request merged**